plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSegmentPageSource.java

```diff
@@ -254,6 +256,8 @@ private long getLong(int rowIndex, int columnIndex)
                 return floatToIntBits(currentDataTable.getDataTable().getFloat(rowIndex, columnIndex));
             case LONG:
                 return currentDataTable.getDataTable().getLong(rowIndex, columnIndex);
+            case TIMESTAMP:
+                return PinotTimestamps.toMicros(currentDataTable.getDataTable().getLong(rowIndex, columnIndex));
```

Comment:
Does Pinot only support timestamps with precision <= 6? If not we should probably add a `verify` here to make sure we don't silently do incorrect things with higher precision values.

Reply:
pinot timestamps are milliseconds, see https://github.com/apache/pinot/blob/master/pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java#L468